### PR TITLE
[ClassLoader][HttpKernel] Remove ClassLoader component & Kernel::loadClassCache

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  |no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/20735
| License       | MIT
| Doc PR        | N/A

As deprecated in https://github.com/symfony/symfony/pull/20735

### CODE REVIEW

**Comment by yceruto** [comment]
I :heart: this part, when we remove stuff and everything continues working fine. Let's go, cleanup, bye to BC layers :smile:

**Comment by ogizanagi** [comment]
@xabbuh : I see you opened #22793. here I did only remove the `ClassCollectionLoader` for the `Kernel::loadClassCache` removal, but I think it make sense to remove everything here. So I cherry-picked your commit here. Is it okay to you ?

**Comment by nicolas-grekas** [comment]
missing CHANGELOG updates in components

**Comment by ogizanagi** [comment]
Added one in HttpKernel: https://github.com/symfony/symfony/pull/22786/files#diff-776debe25c0a772039d45b8d692c9dbfR7. Thanks.

File: src/Symfony/Component/HttpKernel/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ CHANGELOG
  * removed `TraceableControllerResolver::getArguments()`
  * removed `ControllerResolver::getArguments()` and the ability to resolve arguments
  * removed the `argument_resolver` service dependency from the `debug.controller_resolver`
+ * removed `Extension::addClassesToCompile()`
```
Comment (xabbuh) [inline src/Symfony/Component/HttpKernel/CHANGELOG.md]: `Extension:: getClassesToCompile()` too

Comment (xabbuh) [inline src/Symfony/Component/HttpKernel/CHANGELOG.md]: `loadClassCache()` and `setClassCache()` of the `Kernel` class have been removed too

**Comment by nicolas-grekas** [comment]
rebase needed

**Comment by ogizanagi** [comment]
Rebased.

**Comment by nicolas-grekas** [comment]
Thank you @ogizanagi.

**Pull request merged**